esrally/utils/config.py

```diff
@@ -0,0 +1,162 @@
+
+
+class CfgESConnectionOptions:
+    """
+    Base Class to help either parsing --target-hosts or --client-options
+    """
+
+    def __call__(self):
+        """Return a list with the options assigned to the 'default' key"""
+        return self.default
+
+    def __getitem__(self, key):
+        """
+        Race expects the cfg object to be subscriptable
+        Just return 'default'
+        """
+        return self.default
+
+    @property
+    def default(self):
+        """Return a list with the options assigned to the 'default' key"""
+        return self.parsed_options["default"]
+
+    @property
+    def all_options(self):
+        """Return a dict with all parsed options"""
+        return self.parsed_options
+
+
+class TargetHosts(CfgESConnectionOptions):
+    def __init__(self, argvalue):
+        self.argname = "--target-hosts"
+        self.argvalue = argvalue
+        self.parsed_options = []
+
+        self.parse_options()
+
+    def parse_options(self):
+        def normalize_to_dict(arg):
+            """
+            Return parsed comma separated host string as dict with "default" key.
+            This is needed to support backwards compatible --target-hosts for single clusters that are not
+            defined as a json string or file.
+            """
+
+            return {"default": _normalize_hosts(arg)}
+
+        self.parsed_options = to_dict(self.argvalue, default_parser=normalize_to_dict)
+
+
+    @property
+    def all_hosts(self):
+        """Return a dict with all parsed options"""
+        return self.all_options
+
+
+class ClientOptions(CfgESConnectionOptions):
+    DEFAULT_CLIENT_OPTIONS = "timeout:60"
+
+    """
+    Convert --client-options arg to a dict.
+    When no --client-options have been specified but multi-cluster --target-hosts are used,
+    apply options defaults for all cluster names.
+    """
+
+    def __init__(self, argvalue, target_hosts=None):
+        self.argname = "--client-options"
+        self.argvalue = argvalue
+        self.target_hosts = target_hosts
+        self.parsed_options = []
+
+        self.parse_options()
+
+
+    def parse_options(self):
+        def normalize_to_dict(arg):
+            """
+            When --client-options is a non-json csv string (single cluster mode),
+            return parsed client options as dict with "default" key
+            This is needed to support single cluster use of --client-options when not
+            defined as a json string or file.
+            """
+
+            return {"default": kv_to_map(arg)}
+
+        if self.argvalue == ClientOptions.DEFAULT_CLIENT_OPTIONS and self.target_hosts != None:
+            # --client-options unset but multi-clusters used in --target-hosts? apply options defaults for all cluster names.
+            self.parsed_options = {cluster_name: kv_to_map([ClientOptions.DEFAULT_CLIENT_OPTIONS])
+                                   for cluster_name, _ in self.target_hosts.all_hosts.items()}
```

Comment:
Simplify to: `for cluster_name in self.target_hosts.all_hosts.keys()`?

Reply:
Of course :) Addressing.